src/holofuel/pages/Inbox/Inbox.js

```diff
@@ -58,31 +68,122 @@ function useFetchCounterparties () {
+      break
+    case VIEW.recent:
+      displayTransactions = recentTransactions
+      break
+    default:
+      displayTransactions = actionableTransactions
+      break
+  }
 
-    {!isTransactionsEmpty && <div styleName='transaction-list'>
-      {transactions.map(transaction => <TransactionRow
-        transaction={transaction}
-        showConfirmationModal={showConfirmationModal}
-        key={transaction.id} />)}
+  const isPendingTransactionsEmpty = isEmpty(actionableTransactions)
+  const isDisplayTransactionsEmpty = isEmpty(displayTransactions)
+  const pageTitle = `Inbox${isPendingTransactionsEmpty ? '' : ` (${actionableTransactions.length})`}`
+
+  // Sort the transactions by (semantic) date
+  const transactionsByDate = groupBy('dateLabel', displayTransactions)
+  const transactionsByDateKeys = Object.keys(transactionsByDate)
+
+  return <PrimaryLayout headerProps={{ title: pageTitle }} inboxCount={actionableTransactions.length}>
+    <Jumbotron
+      className='inbox-header'
+      title={`${presentHolofuelAmount(holofuelBalance)} HF`}
+      titleSuperscript='Balance'isTransactionsEmpty
+    >
+      <Button styleName='new-transaction-button' onClick={() => showConfirmationModal()}>
+        {/* TODO: Resolve issue with path for the ADD icon >> not displayed properly */}
+        <AddIcon styleName='add-icon' color='#0DC39F' />
+        <h3 styleName='button-text'>New Transaction</h3>
+      </Button>
+
+      <div>
+        {toggleButtons.map(button =>
+          <Button
+            variant={button.view === inboxView ? 'toggle-selected' : 'toggle'}
+            onClick={() => setInboxView(VIEW[button.view])}
+            className={cx(`${button.view}-button`)} /* eslint-disable-line quote-props */
+            key={button.view}>
+            {button.label}
+          </Button>)}
+      </div>
+    </Jumbotron>
+
+    {isEmpty(displayTransactions) && <>
+      <PageDivider title='Today' />
+      <NullStateMessage
+        styleName='null-state-message'
+        message={inboxView === VIEW.pending
+          ? 'You have no pending offers or requests'
+          : 'You have no recent activity'}>
+        <div onClick={() => showConfirmationModal()}>
+          {/* TODO: Resolve issue with path for the ADD icon >> not displayed properly */}
+          <AddIcon styleName='add-icon' color='#0DC39F' />
+          {/* TODO: Remove once the above ADD Icon works... */}
+          <p style={{ fontSize: 30 }}>+</p>
+        </div>
+      </NullStateMessage>
+    </>}
+
+    {!isDisplayTransactionsEmpty && !isEmpty(transactionsByDateKeys) && <div className='transaction-by-date-list'>
+      {transactionsByDateKeys.map(transactionDate => <div key={transactionDate}>
+        <PageDivider title={transactionDate} />
+        <div styleName='transaction-list'>
+          {transactionsByDate[transactionDate].map(transaction => <TransactionRow
+            transaction={transaction}
+            actionsVisible={actionsVisible}
```

Comment:
couple of things. 
First we should be storing just the transaction id in actionsVisible, rather than the whole transaction. It's just safer and cleaner to compare ids than to compare references to objects.

Second I would do the comparison here

`showActionButtons={actionsVisible === transaction.id}`

and then TransactionRow doesn't need to know how the parent component manages which transactions are showing action buttons. This makes the code easier to maintain, refactor, and make feature changes..

Reply:
I still think that `actionsVisible` shouldn't be passed into `TransactionRow`, that instead you should pass in a `showActionButtons` prop so that `TransactionRow` doesn't have to know how its parent handles which transactions are shown.

Reply:
Also, sorry I should have thought of this in my first review: `actionsVisible` should be renamed `actionsVisibleId` to be clear that it's an id.